src/circuit/gadget/sinsemilla.rs

```diff
@@ -194,20 +194,20 @@ where
 
```

Comment:
It would be simpler to constrain a message piece to at most `ceil(C::Base::CAPACITY / K)` bits here, rather than a check that nearly does that here and then a stricter one in `to_base_field`.
```
        // Message piece must be at most `ceil(C::Base::CAPACITY / K)` bits
        let piece_max_num_words = C::Base::CAPACITY as usize / K;
```